### PR TITLE
More backend validations

### PR DESCRIPTION
**Pull request by dreamfall**

### CODE REVIEW

File: lib/qae_form_builder/award_holder_question.rb [hunk 1]
```diff
@@ -1,5 +1,30 @@
 class QAEFormBuilder
   class AwardHolderQuestionValidator < QuestionValidator
+    def errors
```
Comment (matthewford) [inline lib/qae_form_builder/award_holder_question.rb]: Assignment Branch Condition size for errors is too high. [44.81/15]

File: lib/qae_form_builder/award_holder_question.rb [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+            result[question.key][index] ||= ""
+            result[question.key][index] << " #{attr.humanize.capitalize} can't be blank."
+          else
+            if attr == "details"
+              limit = question.delegate_obj.details_words_max
+              length = award[attr].split(" ").length
+
+              if limit && limit_with_buffer(limit) && length && length > limit_with_buffer(limit)
+                result[question.key][index] ||= ""
+                result[question.key][index] << " #{attr.humanize.capitalize} exeeded #{limit} words limit."
```
Comment (matthewford) [inline lib/qae_form_builder/award_holder_question.rb]: Line is too long. [107/100]

File: lib/qae_form_builder/by_trade_goods_and_services_label_question.rb [hunk 3]
```diff
@@ -1,5 +1,30 @@
 class QAEFormBuilder
   class ByTradeGoodsAndServicesLabelQuestionValidator < QuestionValidator
+    HUMANIZED_ATTRS = {
+      "desc_short" => "Description"
+    }
+
+    def errors
```
Comment (matthewford) [inline lib/qae_form_builder/by_trade_goods_and_services_label_question.rb]: Assignment Branch Condition size for errors is too high. [28.23/15]

File: lib/qae_form_builder/multi_question_validator.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+class QAEFormBuilder::MultiQuestionValidator < QAEFormBuilder::QuestionValidator
+  def errors
```
Comment (matthewford) [inline lib/qae_form_builder/multi_question_validator.rb]: Assignment Branch Condition size for errors is too high. [23.22/15]

File: lib/qae_form_builder/question.rb [hunk 5]
```diff
@@ -36,6 +42,14 @@ def skip_base_validation?
         question.delegate_obj.class.name.demodulize == klass
       end
     end
+
+    def limit_with_buffer(limit)
+      if limit > 15
+        limit_with_buffer = (limit + limit * 0.1).to_i + 1
```
Comment (matthewford) [inline lib/qae_form_builder/question.rb]: Useless assignment to variable - `limit_with_buffer`.

File: lib/qae_form_builder/supporters_question.rb [hunk 6]
```diff
@@ -1,5 +1,34 @@
 class QAEFormBuilder
   class SupportersQuestionValidator < QuestionValidator
+    MIN_LIMIT = 2
+
+    def errors
```
Comment (matthewford) [inline lib/qae_form_builder/supporters_question.rb]: Assignment Branch Condition size for errors is too high. [17.55/15]

File: lib/qae_form_builder/textarea_question.rb [hunk 7]
```diff
@@ -3,13 +3,12 @@ class TextareaQuestionValidator < QuestionValidator
...
       result = super
 
-      limit = question.delegate_obj.words_max
-
-      limit_with_buffer = limit
-      limit_with_buffer = (limit + limit * 0.1).to_i + 1 if limit > 15
       length = question.input_value && question.input_value.split(" ").length
 
-      if limit_with_buffer && length && length > limit_with_buffer
+
```
Comment (matthewford) [inline lib/qae_form_builder/textarea_question.rb]: Extra blank line detected.

**Comment by jaleszek** [comment]
After quick look it's fine to me, maybe @RuslanHamidullin wants to take a look as is more competent in this domain.

**Comment by RuslanHamidullin** [comment]
yeah,gonna take a look

**Comment by RuslanHamidullin** [comment]
Looks good :+1:

**Pull request merged**